gst/tensor_merge/gsttensormerge.c

```diff
@@ -0,0 +1,777 @@
+  }
+
+  tensor_merge->tensors_config.rate_d = old_denominator;
+  tensor_merge->tensors_config.rate_n = old_numerator;
+
+  debug_print (!tensor_merge->silent, "pts %" GST_TIME_FORMAT,
+      GST_TIME_ARGS (*pts_time));
+  debug_print (!tensor_merge->silent, "dts %" GST_TIME_FORMAT,
+      GST_TIME_ARGS (*dts_time));
+
+  /* set timestamp */
+  GST_BUFFER_PTS (tensors_buf) = *pts_time;
+  GST_BUFFER_DTS (tensors_buf) = *dts_time;
+  return isEOS;
+}
+
+/**
+ * @brief Gst Collect Pads Function which is called once collect pads done.
+ * @param pads GstCollectPads
+ * @param tensor_merge Merger
+ * @return GstFlowReturn
+ */
+static GstFlowReturn
+gst_tensor_merge_collected (GstCollectPads * pads,
+    GstTensorMerge * tensor_merge)
+{
+  GstFlowReturn ret = GST_FLOW_OK;
+  GstBuffer *tensors_buf, *tensor_buf;
+  GstMemory *mem;
+  GstClockTime pts_time = GST_CLOCK_TIME_NONE;
+  GstClockTime dts_time = GST_CLOCK_TIME_NONE;
+  GstClockTime time = 0;
+  gboolean isEOS = FALSE;
+  GST_DEBUG_OBJECT (tensor_merge, " all pads are collected ");
+  if (tensor_merge->need_stream_start) {
+    gchar s_id[32];
+    g_snprintf (s_id, sizeof (s_id), " tensormerge - %08x ", g_random_int ());
+    gst_pad_push_event (tensor_merge->srcpad,
+        gst_event_new_stream_start (s_id));
+    tensor_merge->need_stream_start = FALSE;
+  }
+
+  tensors_buf = gst_buffer_new ();
+  tensor_buf = gst_buffer_new ();
+
+  isEOS =
+      gst_tensor_merge_collect_buffer (tensor_merge, tensors_buf, &pts_time,
+      &dts_time);
+
+  if (isEOS) {
+    if (tensors_buf)
+      gst_buffer_unref (tensors_buf);
+
+    if (tensor_buf)
+      gst_buffer_unref (tensor_buf);
+
+    gst_pad_push_event (tensor_merge->srcpad, gst_event_new_eos ());
+    ret = GST_FLOW_EOS;
+    goto beach;
+  }
+
+  if (!tensor_merge->negotiated) {
+    GstCaps *newcaps;
+    GstTensorConfig config;
+
+    if (gst_merge_tensors_config (tensor_merge, &tensor_merge->tensors_config,
+            &config)) {
+
+      g_assert (gst_tensor_config_validate (&config));
+      newcaps = gst_tensor_caps_from_config (&config);
+    } else {
+      goto nego_error;
+    }
+
+    if (!gst_pad_set_caps (tensor_merge->srcpad, newcaps)) {
+      gst_caps_unref (newcaps);
+      goto nego_error;
+    }
+
+    gst_caps_unref (newcaps);
+    tensor_merge->negotiated = TRUE;
+  }
+
+  if (tensor_merge->need_segment) {
+    GstSegment segment;
+
+    if (GST_CLOCK_TIME_IS_VALID (dts_time)) {
+      time = dts_time;
+    } else if (GST_CLOCK_TIME_IS_VALID (pts_time)) {
+      time = pts_time;
+    } else {
+      time = 0;
+    }
+
+    gst_segment_init (&segment, GST_FORMAT_TIME);
+    segment.start = time;
+    gst_pad_push_event (tensor_merge->srcpad, gst_event_new_segment (&segment));
+    tensor_merge->need_segment = FALSE;
+  }
+
+
+  /* TODO : Should handle Properly. This is just for the temperaly usage */
+  mem = gst_buffer_get_memory_range (tensors_buf, 0, -1);
+  gst_buffer_append_memory (tensor_buf, mem);
+
```

Comment:
Please copy timestamp from tensors_buf after appending memory block.

gst_buffer_copy_into (tensor_buf, tensors_buf, GST_BUFFER_COPY_TIMESTAMPS, 0, -1);

Reply:
Also, add code to release tensors_buf. 

gst_buffer_unref (tensors_buf);